Benchmarks/Benchmarks/MemcacheBenchmarks/MemcacheBenchmarks.swift

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Benchmark
+import Foundation
+import Memcache
+import NIOCore
+import NIOPosix
+
+let benchmarks = {
+    let defaultMetrics: [BenchmarkMetric] = [
+        .mallocCountTotal,
+        .mallocCountLarge,
+        .mallocCountTotal,
+        .memoryLeaked,
+        .allocatedResidentMemory,
+    ]
+
+    Benchmark("Set Request", configuration: .init(metrics: defaultMetrics)) { benchmark in
+        try await withThrowingTaskGroup(of: Void.self) { group in
+
+            let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
Let's pass the EL to those new functions and let's statically initialize an EL outside of the `let benchmarks`. Take a look at what we do in NIO for it.